sdjournal/journal.go

```diff
@@ -835,6 +837,7 @@ func (j *Journal) GetMonotonicUsec() (uint64, error) {
 // GetCursor gets the cursor of the current journal entry.
 func (j *Journal) GetCursor() (string, error) {
 	var d *C.char
+	defer C.free(unsafe.Pointer(d))
```

Comment:
Moved it down below the `getFunction` call


On 08/23/2016 10:04 AM, Luca Bruno wrote:
>
> In sdjournal/journal.go 
> <https://github.com/coreos/go-systemd/pull/194#discussion_r75885258>:
>
> > @@ -835,6 +837,7 @@ func (j *Journal) GetMonotonicUsec() (uint64, error) {
> >  // GetCursor gets the cursor of the current journal entry.
> >  func (j *Journal) GetCursor() (string, error) {
> >  	var d *C.char
> > +	defer C.free(unsafe.Pointer(d))
>
> This (together with the var declaration perhaps?) should be moved past 
> the |getFunction| call, or it may be freeing a pointer that has never 
> gone through a malloc.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/coreos/go-systemd/pull/194/files/25b29f2e8a8b8dd9d91e37dffaa35fb78ca9e382#r75885258>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABP5GdunjuHbH1_9u7wckRIcFcJvvfUzks5qiwvlgaJpZM4JqjNN>.
>